Sofa/Component/StateContainer/src/sofa/component/statecontainer/MechanicalObject.cpp

```diff
@@ -210,34 +210,14 @@ void MechanicalObject<defaulttype::Rigid3Types>::draw(const core::visual::Visual
             vparams->drawTool()->multMatrix( glTransform );
```

Comment:
why not having out of the loop:
```cpp
auto dt = vparams->drawTool() ;
```
and 
```cpp
auto isSleeping = getContext()->isSleeping();
```

Reply:
yes thanks. Actually, I started to move all the drawing parts into a dedicated component. My intention is to remove the drawing from MechanicalObject. So I will close this PR.